gamla/graph.py

```diff
@@ -8,29 +8,63 @@
 from gamla import functional, functional_generic
 
 
+@toolz.curry
+def _is_better_node(node: Any, seen: Dict, key: Callable):
+    old_node = seen[key(node)]
+    return node > old_node
+
+
+@toolz.curry
+def _save_to_dict(element: Any, dictionary: Dict, func: Callable):
+    dictionary[func(element)] = element
+
+
+@toolz.curry
+def _accumulate(
+    element: Any, is_seen: Callable, better_node: Callable, keep_best: bool
+):
+    return not is_seen(element) or (keep_best and better_node(element))
+
+
 @toolz.curry
 def graph_traverse(
-    source: Any, get_neighbors: Callable, key: Callable = toolz.identity
+    source: Any,
+    get_neighbors: Callable,
+    key: Callable = toolz.identity,
+    keep_best: bool = False,
 ) -> Iterable:
-    yield from graph_traverse_many([source], get_neighbors=get_neighbors, key=key)
+    seen_dict: Dict = {}
+    remember = _save_to_dict(dictionary=seen_dict, func=key)
+    is_seen = toolz.compose_left(key, seen_dict.__contains__)
+    better_node = _is_better_node(seen=seen_dict, key=key)
+
+    yield from graph_traverse_many(
+        [source],
+        get_neighbors=get_neighbors,
+        remember=remember,
+        accumulator=_accumulate(
+            is_seen=is_seen, better_node=better_node, keep_best=keep_best
+        ),
+    )
 
 
 @toolz.curry
 def graph_traverse_many(
-    sources: Any, get_neighbors: Callable, key: Callable = toolz.identity
```

Comment:
to avoid breaking targets,keep this signature and name, and create a new name for the generalized version (make this one use the generic one)
this will allow you to keep the test unskipped as well